types/express-serve-static-core/index.d.ts

```diff
@@ -54,7 +55,7 @@ export type RequestHandlerParams<P extends Params = ParamsDictionary, ResBody =
     | Array<RequestHandler<P>
     | ErrorRequestHandler<P>>;
 
-export interface IRouterMatcher<T> {
+export interface IRouterMatcher<T, Method extends 'all' | 'get' | 'post' | 'put' | 'delete' | 'patch' | 'options' | 'head' = any> {
```

Comment:
Is it possible to create routes for non-standard HTTP methods? If so, then this constraint is too narrow.

Reply:
Poke @andoshin11 / @dwrss  ^

Reply:
@rbuckton Not through the helper methods attached to a router/application, as far as I’m aware (they’re defined [here](https://github.com/expressjs/express/blob/e1b45ebd050b6f06aa38cda5aaf0c21708b0c71e/lib/application.js#L472)).

Reply:
Thanks @dwrss - 👍 works for me, merging